tests/test_run.py

```diff
@@ -44,15 +45,85 @@ def test_default_sends_verify_true():
     resp = Response()
     resp.status_code = 200
     httsleep = HttSleeper(URL, {'status_code': 200})
-    with mock.patch('requests.sessions.Session.send') as mock_session_send:
-        mock_session_send.return_value = resp
+    with mock.patch('requests.adapters.HTTPAdapter.send') as mock_adapter_send:
+        mock_adapter_send.return_value = resp
         httsleep.run()
-        assert mock_session_send.called
-        args, kwargs = mock_session_send.call_args
+        assert mock_adapter_send.called
+        args, kwargs = mock_adapter_send.call_args
     assert 'verify' in kwargs
     assert kwargs['verify'] == True
 
 
+@httpretty.activate
+def test_default_uses_default_session():
+    """Should use a default Session object unless one is specified"""
+    resp = Response()
+    resp.status_code = 200
+    httsleep = HttSleeper(URL, {'status_code': 200})
+    with mock.patch('requests.adapters.HTTPAdapter.send') as mock_adapter_send:
+        mock_adapter_send.return_value = resp
+        httsleep.run()
+        assert mock_adapter_send.called
+        args, kwargs = mock_adapter_send.call_args
+    assert args[0].headers == requests.utils.default_headers()
+    assert 'verify' in kwargs
+    assert kwargs['verify'] == True
+
+
+@httpretty.activate
+def test_propagate_session():
+    """Should propagate a Session's headers, verify setting, cookies etc. when specified"""
+    session = requests.Session()
+    session.cookies = {'tasty-cookie': 'chocolate'}
+    session.headers = {'Content-Type': 'test/type'}
+    session.verify = '/session/verify'
```

Comment:
Should be a boolean, right?

Reply:
Can be either a boolean or a string also - see https://github.com/requests/requests/blob/master/requests/api.py#L39-L41.

I thought it was worth having at least one or two non-boolean cases in the tests.

Reply:
Aha! I wasn't aware. OK, it's grand so